conans/test/functional/command/install_test.py

```diff
@@ -581,3 +583,33 @@ def install_anonymous_test(self):
         client2 = TestClient(servers=servers, users={})
         client2.run("install Pkg/0.1@lasote/testing")
         self.assertIn("Pkg/0.1@lasote/testing: Package installed", client2.out)
+
+    def install_without_ref_test(self):
+        server = TestServer(users={"user": "password"}, write_permissions=[("*/*@*/*", "*")])
+        servers = {"default": server}
+        client = TestClient(servers=servers, users={"default": [("user", "password")]})
+
+        conanfile = textwrap.dedent("""
+                from conans import ConanFile
+
+                class MyPkg(ConanFile):
+                    name = "lib"
+                    version = "1.0"
+                """)
+        client.save({"conanfile.py": conanfile})
+
+        client.run('create .')
+        self.assertIn("lib/1.0: Package '{}' created".format(NO_SETTINGS_PACKAGE_ID),
+                      client.out)
+
+        client.run('upload lib/1.0 -c --all')
+        self.assertIn("Uploaded conan recipe 'lib/1.0' to 'default'", client.out)
+
+        client.run('remove "*" -f')
+
+        # This fails, Conan thinks this is a path
+        client.run('install lib/1.0', assert_error=True)
```

Comment:
Check the error message is something reasonably decent? Might be useless and confusing